packages/config/src/Config.ts

```diff
@@ -0,0 +1,666 @@
+                        break;
+                    }
+                    obj = obj[segment];
+                }
+            }
+
+            // Clear the entry and rebuild it
+            delete this._secure.configs[layer.path];
+
+            // Create the entry to set the secure properties
+            if (Object.keys(sp).length > 0) {
+                this._secure.configs[layer.path] = sp;
+            }
+        }
+
+        // Save the entries if needed
+        if (Object.keys(this._secure.configs).length > 0 || beforeLen > 0 ) {
+            await this._vault.save(Config.SECURE_ACCT, JSONC.stringify(this._secure.configs));
+        }
+    }
+
+    // TODO Does this need to recurse up through nested profiles?
+    private loadProfile(path: string): IConfigLoadedProfile {
+        const profile = lodash.get(this.properties, path);
+        if (profile == null) {
+            return null;
+        }
+
+        const loadedProfile = require("lodash-deep").deepMapValues(profile, (value: any, p: string) => {
+            if (p.includes("properties.")) {
+                for (const layer of this._layers) {
+                    const propertyPath = `${path}.${p}`;
+                    if (lodash.get(layer.properties, propertyPath) != null) {
+                        const property: IConfigLoadedProperty = {
+                            value,
+                            secure: layer.properties.secure.includes(propertyPath),
+                            user: layer.user,
+                            global: layer.global
+                        };
+                        return property;
+                    }
+                }
+            }
+            return value;
+        });
+
+        for (const layer of this._layers) {
+            for (const secureProp of layer.properties.secure) {
+                if (secureProp.startsWith(`${path}.`)) {
+                    const subpath = secureProp.slice(path.length + 1);
+                    if (lodash.get(loadedProfile, subpath) == null) {
+                        lodash.set(loadedProfile, subpath, { secure: true, user: layer.user, global: layer.global });
+                    }
+                }
+            }
+        }
+
+        return loadedProfile;
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    ////////////////////////////////////////////////////////////////////////////
+    // Static Utilities
+    ////////////////////////////////////////////////////////////////////////////
+    ////////////////////////////////////////////////////////////////////////////
+
+    public static search(file: string, opts?: any): string {
+        opts = opts || {};
+        if (opts.stop) opts.stop = node_path.resolve(opts.stop);
+        if (opts.gbl) opts.gbl = node_path.resolve(opts.gbl);
+        const p = findUp.sync((directory: string) => {
+            if (directory === opts.stop) return findUp.stop;
+            if (directory === opts.gbl) return;
```

Comment:
should this be a `findUp.stop` too?

Reply:
No I believe this is deliberate. Should probably have added a comment here though 😢

If we matched the `opts.stop` directory, we want to return the directory we found and look no further.
If we matched the `opts.gbl` directory, we want to return `null` - there is no local config to load if it's in the same folder as global config.

Reply:
if i have home as `C:\Users\dan\` and global as `C:\Users\dan\.zowe`, then I run my project from `C:\Users\dev\` do we just search to root?

Reply:
Yes that's correct (had to test this 😋)

Reply:
thanks! 

i'm curious on the reasoning then - is the thought we stop sooner in "most" cases?  (i'm not saying it's a bad idea; i'm just trying to understand)

Reply:
I'm open to changing it 😁 

@tucker01 Could you explain the reasoning behind stopping in user home folder? As opposed to other tools like NPM that search for node_modules all the way up to the root of your drive.

Reply:
No idea, feel free to change it